sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,394 @@
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(IER12IntegrationScheduler);
+    // Default program year prefix.
+    sharedProgramYearPrefix = 2000;
+    referenceSubmissionDate = new Date("2000-06-01");
+    // Intercept file timestamp.
+    getFileNameAsCurrentTimestampMock = jest.spyOn(
+      dateUtils,
+      "getFileNameAsCurrentTimestamp",
+    );
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    // Update all applications to Overwritten.
+    await db.application.update(
+      { applicationStatus: Not(ApplicationStatus.Overwritten) },
+      { applicationStatus: ApplicationStatus.Overwritten },
+    );
+    // Avoid issues for queries relying on the application number (e.g. hasMultipleApplicationSubmissions).
+    await db.application.update(
+      { applicationNumber: defaultApplicationNumber },
+      { applicationNumber: "0000000000" },
+    );
+  });
+
+  it("Should generate an IER12 file with two records for a single student with no dependents and two disbursements, one sent and one pending.", async () => {
+    // Arrange
+    const testInputData = {
+      student: JOHN_DOE_FROM_CANADA,
+      application: {
+        applicationNumber: defaultApplicationNumber,
+        studentNumber: "A1B2C3D4",
+        relationshipStatus: RelationshipStatus.Single,
+        applicationStatus: ApplicationStatus.Completed,
+        applicationStatusUpdatedOn: undefined,
+      },
+      assessment: {
+        triggerType: AssessmentTriggerType.OriginalAssessment,
+        assessmentDate: undefined,
+        workflowData: WORKFLOW_DATA_SINGLE_INDEPENDENT_WITH_NO_DEPENDENTS,
+        assessmentData: ASSESSMENT_DATA_SINGLE_INDEPENDENT,
+        disbursementSchedules: [
+          {
+            coeStatus: COEStatus.completed,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+            disbursementDate: undefined,
+            updatedAt: undefined,
+            dateSent: undefined,
+            disbursementValues: AWARDS_ONE_OF_TWO_DISBURSEMENT,
+          },
+          {
+            coeStatus: COEStatus.required,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+            disbursementDate: undefined,
+            updatedAt: undefined,
+            dateSent: undefined,
+            disbursementValues: AWARDS_TWO_OF_TWO_DISBURSEMENT,
+          },
+        ],
+      },
+      educationProgram:
+        PROGRAM_UNDERGRADUATE_CERTIFICATE_WITHOUT_INSTITUTION_PROGRAM_CODE,
+      offering: OFFERING_FULL_TIME,
+    };
+    const application = await saveIER12TestInputData(db, testInputData, {
+      programYearPrefix: sharedProgramYearPrefix,
+      submittedDate: referenceSubmissionDate,
+    });
+
+    // Queued job.
+    const job = createIER12SchedulerJobMock(
```

Comment:
as per https://github.com/bcgov/SIMS/pull/2461/files#r1378245645 , we can use job mock without date.

Reply:
We can use it without a date, but the TC is controlling the dates.

Reply:
Are we going to have a TC in upcoming PR, where the TC happens without controlling the date?

Reply:
Maybe if the time allows it. The idea is to use the "Has full-time disbursement feedback errors" where we can set the feedback date to yesterday to test the different statuses.
My plan is to use the rest of the sprint to add scenarios but keep in mind the below agreement persisted to the ticket.
```
Cover a select variety of scenarios, but do not need to have complete coverage of all scenarios.
```
Please keep sharing the ideas and let's see what we can have done this sprint. We can create a ticket for a second effort and register the suggested scenarios not covered in this effort. Hope that it would be ok 😉